OsmAnd/src/net/osmand/plus/quickaction/actions/NavVoiceAction.java

```diff
@@ -31,9 +33,13 @@ public NavVoiceAction(QuickAction quickAction) {
 	}
 
 	@Override
-	public void execute(@NonNull MapActivity mapActivity) {
-		boolean mute = mapActivity.getMyApplication().getSettings().VOICE_MUTE.get();
-		mapActivity.getMyApplication().getSettings().VOICE_MUTE.set(!mute);
+	public void execute(@NonNull MapActivity mapActivity, @Nullable KeyEvent event) {
+		boolean oldMuteState = mapActivity.getMyApplication().getSettings().VOICE_MUTE.get();
+		mapActivity.getMyApplication().getSettings().VOICE_MUTE.set(!oldMuteState);
+
+		if (!oldMuteState) {
+			mapActivity.getMyApplication().getPlayer().stop();
```

Comment:
Probably better to add a listener to this preference in VoiceRouter & stop player there